comp/core/workloadmeta/impl/store_test.go

```diff
@@ -781,6 +781,143 @@ func TestGetProcess(t *testing.T) {
+		expectedError     error
+	}{
+		{
+			description: "process with container",
+			processData: []*wmdef.Process{
+				{
+					EntityID: wmdef.EntityID{
+						Kind: wmdef.KindProcess,
+						ID:   "123",
+					},
+					Owner: &wmdef.EntityID{
+						Kind: wmdef.KindContainer,
+						ID:   "container_id1",
+					},
+				},
+				{
+					EntityID: wmdef.EntityID{
+						Kind: wmdef.KindProcess,
+						ID:   "234",
+					},
+					Owner: &wmdef.EntityID{
+						Kind: wmdef.KindContainer,
+						ID:   "container_id2",
+					},
+				},
+			},
+			containerData: []*wmdef.Container{
+				{
+					EntityID: wmdef.EntityID{
+						Kind: wmdef.KindContainer,
+						ID:   "container_id1",
+					},
+				},
+				{
+					EntityID: wmdef.EntityID{
+						Kind: wmdef.KindContainer,
+						ID:   "container_id2",
+					},
+				},
+			},
+			pidToQuery: "123",
+			expectedContainer: &wmdef.Container{
+				EntityID: wmdef.EntityID{
+					Kind: wmdef.KindContainer,
+					ID:   "container_id1",
+				},
+			},
+		},
+		{
+			description: "process with no container id",
+			processData: []*wmdef.Process{
+				{
+					EntityID: wmdef.EntityID{
+						Kind: wmdef.KindProcess,
+						ID:   "123",
+					},
+				},
+			},
+			containerData:     []*wmdef.Container{},
+			pidToQuery:        "123",
+			expectedContainer: nil,
+			expectedError:     errors.NewNotFound("123"),
+		},
+		{
+			description: "process and container does not exist",
+			processData: []*wmdef.Process{
+				{
+					EntityID: wmdef.EntityID{
+						Kind: wmdef.KindProcess,
+						ID:   "123",
+					},
+					Owner: &wmdef.EntityID{
+						Kind: wmdef.KindContainer,
+						ID:   "container_id1",
+					},
+				},
+			},
+			containerData:     []*wmdef.Container{},
+			pidToQuery:        "123",
+			expectedContainer: nil,
+			expectedError:     errors.NewNotFound("container_id1"),
+		},
+		{
+			description:   "process does not exist",
+			processData:   []*wmdef.Process{},
+			containerData: []*wmdef.Container{},
+			pidToQuery:    "123",
+			expectedError: errors.NewNotFound("process"),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			s := newWorkloadmetaObject(t)
+
+			// store process data into wlm
+			for _, proc := range tc.processData {
+				s.handleEvents([]wmdef.CollectorEvent{
+					{
+						Type:   wmdef.EventTypeSet,
+						Source: fooSource,
+						Entity: proc,
+					},
+				})
+			}
+
+			// store container data into wlm
+			for _, container := range tc.containerData {
+				s.handleEvents([]wmdef.CollectorEvent{
+					{
+						Type:   wmdef.EventTypeSet,
+						Source: fooSource,
+						Entity: container,
+					},
+				})
+			}
+			container, err := s.GetContainerForProcess(tc.pidToQuery)
+			if tc.expectedError != nil {
+				assert.EqualError(t, err, tc.expectedError.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tc.expectedContainer == nil {
+				assert.Nil(t, container)
+			} else {
+				assert.Equal(t, *tc.expectedContainer, *container)
+			}
```

Comment:
Why bother checking the container if it's nil? 

I think you can simplify this to the following and also put in the `require.NoError` when you don't expect it to avoid a panic in case an error does return.

```suggestion
			if tc.expectedError != nil {
				assert.EqualError(t, err, tc.expectedError.Error())
			} else {
				require.NoError(t, err)
				assert.Equal(t, *tc.expectedContainer, *container)
			}
```

Reply:
I had the previous structure because if the expectedContainer is nil, I can't dereference the actual nil container either.

Reply:
Sorry, not quite sure what you mean by this. If there are no errors then `expectedContainer` shouldn't be nil right? `nil` is only returned if there is an error.

Reply:
Ah yes I gotcha. Yeah I can simplify to that. Thanks